### PR TITLE
Observer integration for Worker apps

### PR DESCRIPTION
**Pull request by SharmaAjay19**

### CODE REVIEW

**Comment by michimune** [comment]
Just want to make sure. Is the GeoMaster data provider object instantiated with the correct region? I could not find a code change updating it.

**Comment by SharmaAjay19** [comment]
> Just want to make sure. Is the GeoMaster data provider object instantiated with the correct region? I could not find a code change updating it.

Geomaster details are read from request headers in the diagnostic request middleware and that is how the data provider context is initialized.
image

So the correct geomaster details will be passed by the client, e.g. AppLens or Diagnose & Solve etc.

**Review by hforeste — APPROVED** [review]

File: src/Diagnostics.RuntimeHost/Models/PostBodyModel.cs [hunk 1]
```diff
@@ -73,4 +73,14 @@ public class BodyValidationResult
...
     }
+
+    public class DiagnosticWorkerAppData
+    {
+        public string WorkerAppName { get; set; }
+        public string Tags { get; set; }
+        public string ResourceGroupName { get; set; }
+        public string SubscriptionName { get; set; }
+        public string KubeEnvironmentName { get; set; }
+        public string GeoMasterInstance { get; set; }
```
Comment (yifguoMSFT) [inline src/Diagnostics.RuntimeHost/Models/PostBodyModel.cs:84]: Just a reminder that we just updated the response structure returned by Observer to
image

**Review by yifguoMSFT — APPROVED** [review]

File: src/Diagnostics.RuntimeHost/Models/PostBodyModel.cs (hunk 1 above)
Comment (SharmaAjay19) [reply src/Diagnostics.RuntimeHost/Models/PostBodyModel.cs:84]: Yes, I will be incorporating these changes soon.

**Review by ShekharGupta1988 — APPROVED** [review]

**Pull request merged**